src/common/commands_parser.rs

```diff
@@ -485,85 +482,81 @@ fn parse_disconnect_timeout(args: &ArgMatches) -> NitroCliResult<Option<u64>> {
 
 /// Parse the list of requested CPU IDs from the command-line arguments.
 fn parse_cpu_ids(args: &ArgMatches) -> NitroCliResult<Option<Vec<u32>>> {
-    let cpu_ids_arg = args.values_of("cpu-ids");
-    match cpu_ids_arg {
-        Some(iterator) => {
-            let mut cpu_ids = Vec::new();
-            for cpu_id in iterator {
-                cpu_ids.push(cpu_id.parse().map_err(|_| {
-                    new_nitro_cli_failure!(
-                        "`cpu-id` is not a number",
-                        NitroCliErrorEnum::InvalidArgument
-                    )
-                    .add_info(vec!["cpu-id", cpu_id])
-                })?);
-            }
-            Ok(Some(cpu_ids))
-        }
-        None => Ok(None),
-    }
+    args.get_many::<String>("cpu-ids")
+        .map(|values| {
+            values
+                .map(|id| {
```

Comment:
looks much rustier now 😀

Reply:
however, ideally that refactoring should be in a separate commit as it's not much related to Clap upgrade